logic_tree_tools.py

```diff
@@ -0,0 +1,972 @@
+
+        if "Subduction Interface" in tectonic_region_type_set:
+
+            for which_interface in which_interfaces:
+
+                logic_tree_pair_for_tectonic_region_type_set = (
+                    logic_tree_pair_with_selected_tectonic_region_types(
+                        initial_logic_tree_pair,
+                        tectonic_region_type_set=tectonic_region_type_set,
+                        which_interface=which_interface,
+                    )[0]
+                )
+
+                logic_tree_pair_list.append(
+                    logic_tree_pair_for_tectonic_region_type_set
+                )
+
+        else:
+
+            logic_tree_pair_for_tectonic_region_type_set = (
+                logic_tree_pair_with_selected_tectonic_region_types(
+                    initial_logic_tree_pair,
+                    tectonic_region_type_set=tectonic_region_type_set,
+                    which_interface=None,
+                )[0]
+            )
+
+            logic_tree_pair_list.append(logic_tree_pair_for_tectonic_region_type_set)
+
+    return logic_tree_pair_list
+
+
+def get_logic_tree_pairs_for_individual_ground_motion_models(
+    initial_logic_tree_pair: CustomLogicTreePair,
+    tectonic_region_type_sets: list[list[str]],
+    which_interfaces: list[str],
+) -> list[CustomLogicTreePair]:
+    """
+    Creates a list of logic tree pairs with all individual ground motion models within the selected tectonic region types.
+
+    Parameters
+    ----------
+    initial_logic_tree_pair : CustomLogicTreePair
+        The initial logic tree pair to select tectonic region types from.
+        Should contain the full SourceLogicTree and the full GMCMLogicTree.
+
+    tectonic_region_type_sets : list[list[str]]
+        A list of lists that each containing a single tectonic region type
+            Valid tectonic region types are:
+            "Active Shallow Crust",
+            "Subduction Interface",
+            "Subduction Intraslab".
+        Examples:
+         [["Active Shallow Crust"], ["Subduction Interface"], ["Subduction Intraslab"]]
+
+    which_interfaces : str, default = "HIK_and_PUY"
+        Which subduction interfaces to include.
+        Valid options are:
+           "HIK_and_PUY" which includes both the Hikurangi–Kermadec and Puysegur subduction zones
+           "only_HIK" which includes only the Hikurangi–Kermadec subduction zone
+           "only_PUY" which includes only the Puysegur subduction zone.
+
+    Returns
+    -------
+    modified_logic_tree_pair_list : list[CustomLogicTreePair]
+        A list of logic tree pairs, each containing only an individual ground motion model.
+
+    Raises
+    ------
+    AssertionError
+        If more than one tectonic_region_type is included in any tectonic_region_type_set.
+    """
+
+    assert (
+        len(tectonic_region_type_sets[0]) == 1
+    ), "Only one tectonic_region_type can be included in each tectonic_region_type_set passed to this function."
```

Comment:
Assert is nice, but assertion errors are not particularly specific. I would change this to a 

```
if len(tectonic_region_types_set[0]) == 1:
    raise ValueError(...)
```